x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/util/Wait.java

```diff
@@ -0,0 +1,139 @@
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isType;
+
+/**
+ * Slowdown function - for debug purposes only.
+ * Syntax: WAIT(ms) - will sleep for ms milliseconds.
+ */
+public class Wait extends UnaryScalarFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Wait", Wait::new);
+
+    @FunctionInfo(
+        returnType = { "boolean" },
+        description = "Debug function to slow down processing. The function will sleep for the specified number of milliseconds.",
+        examples = { @Example(file = "null", tag = "wait") }
+    )
+    public Wait(
+        Source source,
+        @Param(name = "ms", type = { "integer", "long", "time_duration", }, description = "For how long") Expression ms
+    ) {
+        super(source, ms);
+    }
+
+    private Wait(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public Expression replaceChildren(List<Expression> newChildren) {
+        return new Wait(source(), newChildren.getFirst());
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    public DataType dataType() {
+        return DataType.BOOLEAN;
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
+        if (childrenResolved() == false) {
+            return new TypeResolution("Unresolved children");
+        }
+
+        return isType(
+            field(),
+            t -> t == DataType.TIME_DURATION || t.isWholeNumber(),
+            sourceText(),
+            FIRST,
+            "long",
+            "integer",
+            "time_duration"
+        );
+    }
+
+    @Override
+    public Nullability nullable() {
+        return Nullability.FALSE;
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, Wait::new, field());
+    }
+
+    @Override
+    public boolean foldable() {
+        return false;
+    }
+
+    @Override
+    public Object fold() {
+        return null;
+    }
+
+    private long msValue() {
+        if (field().foldable() == false) {
+            throw new IllegalArgumentException("function [" + sourceText() + "] has invalid argument [" + field().sourceText() + "]");
+        }
+        var ms = field().fold();
+        if (ms instanceof Duration duration) {
+            return duration.toMillis();
+        }
+        return ((Number) ms).longValue();
+    }
+
+    @Override
+    public ExpressionEvaluator.Factory toEvaluator(EvaluatorMapper.ToEvaluator toEvaluator) {
+        return new WaitEvaluator.Factory(source(), msValue());
+    }
+
+    @Evaluator
+    static boolean process(@Fixed long ms) {
+        // Only activate in snapshot builds
+        if (Build.current().isSnapshot()) {
+            try {
+                Thread.sleep(ms);
+            } catch (InterruptedException e) {
+                return true;
+            }
```

Comment:
sleep vs wait have very different semantics. What exactly are you trying to achieve?
The name has to be aligned with the implementation - doing a Wait which does Sleep is confusing.

Reply:
I am trying to delay the execution of the query for a bit, to simulate slow query. No additional semantics implied.

Reply:
The call it delay.

Reply:
OK delay sounds good.